File: frontend/src/api/uiScriptTemplate.ts
import request from './request'
import type { ApiResult } from './types'
import type { PageResult } from './testReport'

export interface UIScriptTemplateRequestDTO {
    templateName: string
    templateType: 'SELENIUM' | 'PLAYWRIGHT' | 'PUPPETEER' | string
    scriptLanguage: 'PYTHON' | 'JAVA' | 'JAVASCRIPT' | string
    templateContent: string
    templateVariables?: string
    applicableScenarios?: string
    templateDescription?: string
    isActive?: string
    creatorId?: number
}

export interface UIScriptTemplateResponseDTO {
    id: number
    templateCode: string
    templateName: string
    templateType: string
    scriptLanguage: string
    templateContent: string
    templateVariables: string
    applicableScenarios: string
    templateDescription: string
    version: number
    isActive: string
    creatorId: number
    createTime: string
    updateTime: string
}

export const uiScriptTemplateApi = {
    // Create template
    createTemplate(data: UIScriptTemplateRequestDTO) {
        return request.post<UIScriptTemplateResponseDTO>('/v1/ui-script-templates', data)
    },

    // Get template list
    getTemplateList(page: number = 0, size: number = 10, templateName?: string, templateType?: string, scriptLanguage?: string, isActive?: string) {
        return request.get<ApiResult<PageResult<UIScriptTemplateResponseDTO>>>('/v1/ui-script-templates', {
            params: { page, size, templateName, templateType, scriptLanguage, isActive }
        })
    },

    // Get template by ID
    getTemplateById(id: number) {
        return request.get<UIScriptTemplateResponseDTO>(`/v1/ui-script-templates/${id}`)
    },

    // Get template by code
    getTemplateByCode(templateCode: string) {
        return request.get<UIScriptTemplateResponseDTO>(`/v1/ui-script-templates/code/${templateCode}`)
    },

    // Update template
    updateTemplate(id: number, data: UIScriptTemplateRequestDTO) {
        return request.put<UIScriptTemplateResponseDTO>(`/v1/ui-script-templates/${id}`, data)
    },

    // Delete template
    deleteTemplate(id: number) {
        return request.delete<void>(`/v1/ui-script-templates/${id}`)
    },

    // Get active templates by type and language
    getActiveTemplates(templateType?: string, scriptLanguage?: string) {
        return request.get<UIScriptTemplateResponseDTO[]>('/v1/ui-script-templates/active', {
            params: { templateType, scriptLanguage }
        })
    },

    // Update template status
    updateTemplateStatus(id: number, isActive: string) {
        return request.put<UIScriptTemplateResponseDTO>(`/v1/ui-script-templates/${id}/status`, null, {
            params: { isActive }
        })
    }
}

File: backend-java/test-design-assistant-core/src/main/java/com/sinosoft/testdesign/entity/UIScriptTemplate.java
package com.sinosoft.testdesign.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * UI脚本模板实体
 * 
 * @author sinosoft
 * @date 2024-01-17
 */
@Data
@Entity
@Table(name = "ui_script_template")
public class UIScriptTemplate {
    
    /**
     * 主键ID
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    /**
     * 模板编码（TMP-YYYYMMDD-序号）
     */
    @Column(name = "template_code", unique = true, nullable = false, length = 100)
    private String templateCode;
    
    /**
     * 模板名称
     */
    @Column(name = "template_name", nullable = false, length = 500)
    private String templateName;
    
    /**
     * 模板类型：SELENIUM/PLAYWRIGHT/PUPPETEER
     */
    @Column(name = "template_type", nullable = false, length = 50)
    private String templateType;
    
    /**
     * 脚本语言：PYTHON/JAVA/JAVASCRIPT
     */
    @Column(name = "script_language", nullable = false, length = 50)
    private String scriptLanguage;
    
    /**
     * 模板内容
     */
    @Column(name = "template_content", nullable = false, columnDefinition = "TEXT")
    private String templateContent;
    
    /**
     * 模板变量定义（JSON格式）
     */
    @Column(name = "template_variables", columnDefinition = "TEXT")
    private String templateVariables;
    
    /**
     * 适用场景描述
     */
    @Column(name = "applicable_scenarios", columnDefinition = "TEXT")
    private String applicableScenarios;
    
    /**
     * 模板描述
     */
    @Column(name = "template_description", columnDefinition = "TEXT")
    private String templateDescription;
    
    /**
     * 版本号
     */
    @Column(name = "version")
    private Integer version = 1;
    
    /**
     * 是否启用：1-启用，0-禁用
     */
    @Column(name = "is_active", length = 1)
    private String isActive = "1";
    
    /**
     * 创建人ID
     */
    @Column(name = "creator_id")
    private Long creatorId;
    
    /**
     * 创建时间
     */
    @Column(name = "create_time")
    private LocalDateTime createTime;
    
    /**
     * 更新时间
     */
    @Column(name = "update_time")
    private LocalDateTime updateTime;
    
    @PrePersist
    public void prePersist() {
        this.createTime = LocalDateTime.now();
        this.updateTime = LocalDateTime.now();
    }
    
    @PreUpdate
    public void preUpdate() {
        this.updateTime = LocalDateTime.now();
    }
}

